### PR TITLE
Suppress

### PR DESCRIPTION
**Pull request by bruntib**
Part of  #490

### CODE REVIEW

File: viewer_clients/cmdline_client/cmd_line_client.py [hunk 1]
```diff
@@ -240,12 +243,23 @@ def add_filter_conditions(report_filter, filter_str):
...
     if path:
         report_filter.filepath = path
 
+
+def add_server_arguments(parser):
+    parser.add_argument('--host', type=str, dest="host",
+                        default='localhost',
+                        help='Server host.')
+    parser.add_argument('-p', '--port', type=str, dest="port",
+                        default=11444,
```
Comment (whisperity) [inline viewer_clients/cmdline_client/cmd_line_client.py]: Though I know this is just a DRY extraction, the default port number `11444` is no longer in line with the default view-port number `8001` set here: https://github.com/Ericsson/codechecker/blob/master/codechecker/CodeChecker.py#L416

**Comment by gyorb** [comment]
New features for this issue #490

File: viewer_clients/cmdline_client/cmd_line_client.py (hunk 1 above)
Comment (gyorb) [inline viewer_clients/cmdline_client/cmd_line_client.py]: I'm cleaning up the cmd_line_client and will change it.

File: api/report_viewer_server.thrift [hunk 2]
```diff
@@ -61,7 +61,8 @@ struct ReportFilter{
   3: optional shared.Severity severity,
   4: optional string          checkerId,          // should filter in the fully qualified checker id name such as alpha.core.
                                                   // the analyzed system. Projects can optionally use this concept.
-  5: optional bool            suppressed = false  // if the bug state is suppressed
+  5: optional bool            suppressed = false, // if the bug state is suppressed
```
Comment (gyorb) [inline api/report_viewer_server.thrift]: API version number should be increased. There are new types and new api calls introduced.

File: libcodechecker/cmd/cmd_line_client.py [hunk 3]
```diff
@@ -464,6 +473,67 @@ def printResult(getterFn, baseid, newid, suppr, output_format):
...
+                out.write(suppression.bug_hash + '||' +
+                          suppression.file_name + '||' +
+                          suppression.comment + '\n')
+
+    elif args.input:
+        with open(args.input, 'r') as inp:
+            for line in inp:
+                bug_id, file_name, comment = line.rstrip().split('||')
+
+                reports = client.getRunResults(run_id, 500, 0, None,
```
Comment (gyorb) [inline libcodechecker/cmd/cmd_line_client.py]: The `500` value should not be hardcoded, there is a constant value in the thrift api files which should be used.

File: libcodechecker/cmd/cmd_line_client.py [hunk 4]
```diff
@@ -599,25 +650,64 @@ def register_client_command_line(argument_parser):
...
                                              'CodeChecker server.')
-    auth_parser.add_argument('--host', type=str, dest="host",
-                             default='localhost', help='Server host.')
-    auth_parser.add_argument('-p', '--port', type=str, dest="port",
-                             default="8001", help='HTTP Server port.')
     auth_parser.add_argument('-u', '--username', type=str, dest="username",
                              required=False,
                              help='Username to use on authentication.',
                              default=getpass.getuser())
+    auth_parser.add_argument('-pw', '--password', type=str, dest="password",
```
Comment (gyorb) [inline libcodechecker/cmd/cmd_line_client.py]: Why is this added back?

File: libcodechecker/cmd/cmd_line_client.py [hunk 5]
```diff
@@ -239,14 +239,23 @@ def add_filter_conditions(report_filter, filter_str):
...
-            shared.ttypes.Severity._NAMES_TO_VALUES[severity.upper()]
+        report_filter.severity\
+            = shared.ttypes.Severity._NAMES_TO_VALUES[severity.upper()]
     if checker:
         report_filter.checkerId = '*' + checker + '*'
     if path:
         report_filter.filepath = path
 
 
+def add_server_arguments(parser):
```
Comment (whisperity) [inline libcodechecker/cmd/cmd_line_client.py]: Please remove this function and changes related to this.

File: libcodechecker/cmd/cmd_line_client.py [hunk 6]
```diff
@@ -474,11 +544,7 @@ def register_client_command_line(argument_parser):
     listruns_parser = subparsers.add_parser('runs',
                                             formatter_class=ADHFormatter,
                                             help='Get the run data.')
-    listruns_parser.add_argument('--host', type=str, dest="host",
-                                 default='localhost',
-                                 help='Server host.')
-    listruns_parser.add_argument('-p', '--port', type=str, dest="port",
-                                 default="8001", help='HTTP Server port.')
+    add_server_arguments(listruns_parser)
```
Comment (whisperity) [inline libcodechecker/cmd/cmd_line_client.py]: Please revert this hunk change, and all such, related to `add_server_arguments` below.

File: libcodechecker/cmd/cmd_line_client.py [hunk 7]
```diff
@@ -599,25 +650,64 @@ def register_client_command_line(argument_parser):
...
+                                 'suppressed. It has sense only when '
+                                 '--bugid is also provided. If this parameter '
+                                 'is not given then all reports with the '
+                                 '--bugid will be suppressed/unsuppressed '
+                                 'regardless of the file.')
+    suppress_parser.add_argument('-n', '--name', type=str, dest='name',
+                                 required=True,
+                                 help='Run name.')
+    logger.add_verbose_arguments(suppress_parser)
+    add_server_arguments(suppress_parser)
```
Comment (whisperity) [inline libcodechecker/cmd/cmd_line_client.py]: **no** need to touch this line, though

Comment (whisperity) [inline libcodechecker/cmd/cmd_line_client.py]: Well, actually, there is, due to the tests.

File: libcodechecker/cmd/cmd_line_client.py (hunk 5 above)
Comment (whisperity) [inline libcodechecker/cmd/cmd_line_client.py]: Because of the tests for suppress, either keep this, or copy the contents into it's call site at `suppress` arguments.

File: libcodechecker/cmd/cmd_line_client.py (hunk 4 above)
Comment (whisperity) [inline libcodechecker/cmd/cmd_line_client.py]: Most likely a rebase mistake.

File: libcodechecker/cmd/cmd_line_client.py (hunk 5 above)
Comment (bruntib) [inline libcodechecker/cmd/cmd_line_client.py]: I still can't see which this has to be reverted.

Comment (whisperity) [inline libcodechecker/cmd/cmd_line_client.py]: I have an upcoming merge request that changes the `cmd` subcommand to the new structure. The change irrelevant to the implementation of `suppress` sub-subcommand makes the previously created diffs a lot harder to manage.

But as you see fit, a rebase would be essential anyways for me later on.

File: libcodechecker/cmd/cmd_line_client.py [hunk 8]
```diff
@@ -465,6 +474,69 @@ def printResult(getterFn, baseid, newid, suppr, output_format):
...
+    already_suppressed = 'Bug {} in file {} already suppressed. Use --force!'
+    limit = codeCheckerDBAccess.constants.MAX_QUERY_SIZE
+
+    client = setupClient(args.host, args.port, '/')
+
+    run_info = check_run_names(client, [args.name])
+    run_id, run_date = run_info.get(args.name)
+
+    if args.output:
+        with open(args.output, 'w') as out:
```
Comment (gyorb) [inline libcodechecker/cmd/cmd_line_client.py]: Writing and reading the suppress file should be done only through the [https://github.com/Ericsson/codechecker/blob/master/libcodechecker/suppress_file_handler.py](suppress_file_handler.py) module. That module handles the reading and writing of the suppress file (the format is specified there).

File: libcodechecker/generic_package_suppress_handler.py [hunk 9]
```diff
@@ -48,7 +48,7 @@ def __revalidate_suppress_data(self):
     def store_suppress_bug_id(self, bug_id, file_name, comment):
 
         if self.suppress_file is None:
-            return False
+            return True
```
Comment (gyorb) [inline libcodechecker/generic_package_suppress_handler.py]: Shouldn't we re validate/update the in memory suppress data here?

File: libcodechecker/generic_package_suppress_handler.py [hunk 10]
```diff
@@ -60,7 +60,7 @@ def store_suppress_bug_id(self, bug_id, file_name, comment):
     def remove_suppress_bug_id(self, bug_id, file_name):
 
         if self.suppress_file is None:
-            return False
+            return True
```
Comment (gyorb) [inline libcodechecker/generic_package_suppress_handler.py]: Shouldn't we re validate/update the in memory suppress data here?

File: libcodechecker/generic_package_suppress_handler.py (hunk 9 above)
Comment (bruntib) [inline libcodechecker/generic_package_suppress_handler.py]: The `__revalidate_suppress_data()` reads the suppression information from the file. This `if` statement is running when this file is missing.

File: libcodechecker/suppress_file_handler.py [hunk 11]
```diff
@@ -46,6 +46,7 @@ def get_suppress_data(suppress_file):
     suppress_data = []
 
     for line in suppress_file:
+        line = line.rstrip()
```
Comment (gyorb) [inline libcodechecker/suppress_file_handler.py]: Why do we need the `rstrip` here? Before matching the regex `strip` is called.
